modules/extension/wmts/src/main/java/org/geotools/ows/wmts/request/AbstractGetTileRequest.java

```diff
@@ -134,6 +134,11 @@ public void setLayer(WMTSLayer layer) {
             StyleImpl defaultStyle = layer.getDefaultStyle();
             if (defaultStyle != null && defaultStyle.getName() != null) {
                 styleName = defaultStyle.getName();
+            } else if (layer.getStyles().size() > 0) {
+                StyleImpl topStyle = layer.getStyles().get(0);
+                if (topStyle.getName() != null) {
```

Comment:
Since you are checking for nulls, maybe this if could become a while loop, and exit once you find the first nonnull name....